### ISSUE TITLE
UIDemo1

### ISSUE THREAD

**Issue by fengyiyangdenanren** [fengyiyangdenanren/WyDemo#13]
@{
    ViewBag.Title = "编辑";
    Layout = null;
    var sexList = ViewBag.SexList as List<MessageBus.Web.Controllers.SexType>;

    Dictionary<string, object> isOpenStyle = new Dictionary<string, object>();
    isOpenStyle.Add("lay-skin", "switch");
}
@model MessageBus.Web.Controllers.UIDemoModel
<div class="layui-tab layui-tab-brief my-tab-title">
    <ul class="layui-tab-title">
        <li class="layui-this"><a href="javascript:;">编辑管理</a></li>
        <div class="my-func-box">
            <a href="javascript:;" href-url="@(ViewBag.ReferrerUrl)" class="layui-btn layui-btn-primary layui-btn-mini btnLoadItem">返回上一页</a>
            <a href="javascript:;" href-url="@(ViewBag.CurrentUrl)" class="layui-btn layui-btn-primary layui-btn-mini btnLoadItem">刷新</a>
        </div>
    </ul>
</div>
<blockquote class="layui-elem-quote layui-quote-nm">
    2、这是用来对一些页面进行注释说明使用的
</blockquote>
@using (Ajax.BeginForm("AjaxEdit", "UIDemo", new AjaxOptions { OnBegin = "AjaxBegin", OnFailure = "AjaxFailure", OnSuccess = "AjaxSuccess" }, new { @class = "layui-form", Id = "formUIDemo" }))
{
    <div class="layui-form-item">
        <label class="layui-form-label">@Html.DisplayNameFor(o => o.UIId)</label>
        <div class="layui-input-block">
            @Html.TextBoxFor(o => o.UIId, new { @class = "layui-input" })
        </div>
    </div>
    <div class="layui-form-item">
        <label class="layui-form-label">@Html.DisplayNameFor(o => o.UIName)</label>
        <div class="layui-input-block">
            @Html.TextBoxFor(o => o.UIName, new { @class = "layui-input" })
        </div>
    </div>
    <div class="layui-form-item">
        <div class="layui-inline">
            <label class="layui-form-label">@Html.DisplayNameFor(o => o.UITime)</label>
            <div class="layui-input-block">
                @Html.TextBoxFor(o => o.UITime, new { @class = "layui-input" })
            </div>
        </div>
        <div class="layui-inline">
            <label class="layui-form-label">@Html.DisplayNameFor(o => o.UIMoney)</label>
            <div class="layui-input-block">
                @Html.TextBoxFor(o => o.UIMoney, new { @class = "layui-input" })
            </div>
        </div>
    </div>
    <div class="layui-form-item">
        <label class="layui-form-label">@Html.DisplayNameFor(o => o.UIStatus)</label>
        <div class="layui-input-block">
            @Html.DropDownListFor(o => o.UIStatus, null)
        </div>
    </div>
    <div class="layui-form-item">
        <label class="layui-form-label">@Html.DisplayNameFor(o => o.IsOpen)</label>
        <div class="layui-input-block">
            @Html.CheckBoxFor(o => o.IsOpen, isOpenStyle)
        </div>
    </div>
    <div class="layui-form-item">
        <div class="layui-inline">
            <label class="layui-form-label">@Html.DisplayNameFor(o => o.Mobile)</label>
            <div class="layui-input-block">
                @Html.TextBoxFor(o => o.Mobile, new { @class = "layui-input" })
            </div>
        </div>
        <div class="layui-inline">
            <label class="layui-form-label">@Html.DisplayNameFor(o => o.Email)</label>
            <div class="layui-input-block">
                @Html.TextBoxFor(o => o.Email, new { @class = "layui-input" })
            </div>
        </div>
    </div>
    <div class="layui-form-item">
        <label class="layui-form-label">@Html.DisplayNameFor(o => o.UIUrl)</label>
        <div class="layui-input-block">
            @Html.TextBoxFor(o => o.UIUrl, new { @class = "layui-input" })
        </div>
    </div>
    <div class="layui-form-item">
        <label class="layui-form-label">@Html.DisplayNameFor(o => o.Sex)</label>
        <div class="layui-input-block">
            @foreach (var sexItem in sexList)
            {
                if (Model.Sex == sexItem)
                {
                    @Html.RadioButtonFor(o => o.Sex, (int)sexItem, new { title = sexItem.ToString(), @checked = "checked" })
                }
                else
                {
                    @Html.RadioButtonFor(o => o.Sex, (int)sexItem, new { title = sexItem.ToString() })
                }
            }
        </div>
    </div>
    <div class="layui-form-item">
        <div class="layui-input-block">
            <button class="layui-btn layui-btn-normal" type="submit"> 保 存 </button>
            <a class="layui-btn layui-btn-primary btnLoadItem" href-url="@(ViewBag.CurrentUrl)"> 取 消 </a>
        </div>
    </div>
}
<script src="~/Content/js/jquery.validate.unobtrusive.min.js"></script>
<script type="text/javascript">
    layui.form().render();
    var isAjaxBegin = false; var indexAjaxBegin;
    function AjaxBegin() {
        if (isAjaxBegin) { return false; } else { isAjaxBegin = true; }

        indexAjaxBegin = layer.load(1);
    }
    function AjaxFailure(data, status) {
        layer.close(indexAjaxBegin);
        layer.msg(data.statusText, { icon: 5, time: 1200, anim: 6 }, function () {

        });
        isAjaxBegin = false;
    }
    function AjaxSuccess(data, status) {
        layer.close(indexAjaxBegin);
        if (data.Status) {
            layer.msg(data.Message, { icon: 6, time: 1200, anim: 6 }, function () {
                //在这做成功处理
            });
        }
        else {
            layer.msg(data.Message, { icon: 5, time: 1200, anim: 6 }, function () {
                //在这做成功处理
            });
        }
        isAjaxBegin = false;
    }

    ErrorTips = function (id, msg) {
        layer.tips(msg, "#" + id, { tips: [1, '#ff5722'], tipsMore: true, time: 10000 });
    }
</script>
